pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -188,11 +189,31 @@ def test_infer_dtype_from_array(arr, expected, pandas_dtype):
         (Period("2011-01-01", freq="D"), object),
     ],
 )
-def test_cast_scalar_to_array(obj, dtype):
-    shape = (3, 2)
-
+@pytest.mark.parametrize("shape", [(), (5,), (3, 2)])
+def test_cast_scalar_to_array(obj, dtype, shape):
     exp = np.empty(shape, dtype=dtype)
     exp.fill(obj)
 
     arr = cast_scalar_to_array(shape, obj, dtype=dtype)
     tm.assert_numpy_array_equal(arr, exp)
+
+
+@pytest.mark.parametrize(
+    "obj_in,dtype_in,obj_out,dtype_out",
+    [
+        (NaT, "datetime64[ns]", np.datetime64("NaT"), "datetime64[ns]"),
+        (Timestamp(1), "datetime64[ns]", 1, "datetime64[ns]"),
```

Comment:
can you add a tz-aware Timestamp case
can you add Period, Interval as well

Reply:
you can probably use this fixture: ea_scalar_and_dtype

Reply:
to be sure: as we cast to np.dtype, Period and Interval should be left aside, right?